src/components/SearchResult.tsx

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import { Box, Container, Link, styled } from "@material-ui/core";
+import Label from "./Label";
+import { makeStyles } from "@material-ui/core/styles";
+
+//This is just a draft, better interface, with descriptions, is gonna be implemented in next PR
+export interface Item {
+  vocabulary: string;
+  description?: string;
+}
+
+export interface SearchItem {
+  label: string;
+  isWord: boolean;
+  vocabularies: string[];
+  items?: Item[];
+}
+
+export interface clickCallback {
+  click: () => void;
+}
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    "&:hover": {
+      cursor: "pointer",
+    },
+  },
+}));
+
+const SearchResult: React.FC<SearchItem & clickCallback> = (props) => {
+  const SearchBox = styled(Box)(({ theme }) => ({
+    borderLeft: "solid",
+    borderColor: props.isWord
+      ? theme.palette.secondary.main
+      : theme.palette.primary.main,
+    paddingLeft: theme.spacing(2),
+    marginTop: theme.spacing(2),
+  }));
+  const classes = useStyles();
+  return (
+    <Container>
+      <SearchBox>
+        <Link
+          classes={classes}
+          color="textPrimary"
+          underline="always"
+          variant="h2"
+          onClick={props.click}
```

Comment:
This absolutely needs to be a `href` attribute instead of an onClick event because of accessibility reasons. At the very least you cannot e.g. open this "link" in a new tab, which is pretty standard modus operandi when working with search results. In addition you cannot distinguish visited and non-visited links.